### PR TITLE
build-all.sh: Initial revision

### PR DESCRIPTION
**Pull request by nathanchance**
This is a script that will run all of the builds that Travis does on a
local user's machine. This can help validate changes faster than Travis
can if the user has a very fast machine (like I do).

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
> if the user has a very fast machine (like I do).

weird flex but :ok_hand:

File: build-all.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    [[ -n ${HOURS} ]] && TIME_STRING="${HOURS}h "
+    echo "${TIME_STRING}${MINS}m ${SECS}s"
+)}
+
+BLD_ALL_START=$(date +%s)
+
+echo
+grep "env:" .travis.yml | grep -v LLVM_VERSION | sed "s/.*env: //g" | while read -r ITEM; do (
+    DRIVER_START=$(date +%s)
+    read -ra VALUES <<< "${ITEM}"
```
Comment (nickdesaulniers) [inline build-all.sh]: `<<<` will fail on dash shell

I learned this the hard way recently when Android kernel builds were failing, but only due to the build machines using `dash` while the internal "glinux" distros used `bash` (and I use `zsh`).

File: build-all.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        export "${VALUE:?}"
+    done
+
+    # Make sure that the version suffix is stripped for local builds,
+    # where it is assumed that the user will be using a tip of tree build
+    [[ -n ${LD} ]] && export LD=${LD//-*}
+
+    echo -e "Running '${ARCH:+ARCH=${ARCH} }${LD:+LD=${LD} }${REPO:+REPO=${REPO} }./driver.sh'... \c"
+    if ! ./driver.sh "${@}" &>/dev/null; then
+        echo -e "\033[01;31mFailed\033[0m\c"
```
Comment (nickdesaulniers) [inline build-all.sh]: please use variables for what I assume is green/red/reset

File: build-all.sh (hunk 1 above)
Comment (nathanchance) [reply build-all.sh]: Hmmm, the shebang specifies `bash` so is this a problem? I'll try to look at a more portable solution though.

File: build-all.sh (hunk 2 above)
Comment (nathanchance) [reply build-all.sh]: Sure.

File: build-all.sh (hunk 1 above)
Comment (nathanchance) [reply build-all.sh]: Well, `dash` doesn't do arrays so this entire script will have to be rewritten if being portable is a requirement. `driver.sh` uses some `bash`-isms too, I don't think it is unreasonable to require `bash` for these scripts. Let me know if you feel strongly about that though :)

**Pull request merged**